libpod/oci_conmon_linux.go

```diff
@@ -1149,8 +1165,10 @@ func (r *ConmonOCIRuntime) configureConmonEnv(runtimeDir string) ([]string, []*o
 	env = append(env, fmt.Sprintf("HOME=%s", home))
 
 	extraFiles := make([]*os.File, 0)
-	if notify, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
-		env = append(env, fmt.Sprintf("NOTIFY_SOCKET=%s", notify))
+	if ctr.config.SdNotifyMode == define.SdNotifyModeContainer {
+		if notify, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
+			env = append(env, fmt.Sprintf("NOTIFY_SOCKET=%s", notify))
```

Comment:
who will ensure the `notify` socket is accessible inside of the container?

Reply:
SdNotifyModeContainer - OCI runtime does
SdNotifyModeConmon/ignore - The notify socket ISN'T accessible in the container.

With this PR, all we do is selectively pass the NOTIFY_SOCKET to the OCI runtime.